### ISSUE TITLE
FEATURE: Service calls for EC entities to control behaviour in from other automations

### ISSUE THREAD

**Issue by danobot** [danobot/entity-controller#125]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Implement support for service calls such that the EC instance itself can be turned on and off using service calls (for control from other automations).

For example, there may a python script monitoring your fish tank with complex logic to determine when heater, pump and light are turned on. (yay for random exmaples)

This script should be able to control the EC instance itself by calling HA services such as `constrain`, `enable`, `return_to_idle` on the EC instance to overwrite what the state machine is doing internally.

**Describe alternatives you've considered**
N/A

**Additional context**
This has been on my backlog for a long time. Mentinoed in [this comment](https://github.com/danobot/entity-controller/issues/106#issuecomment-567925985)

**Comment by xannor** [comment]
I have an idea about this and I wanted to run it by you before I attempt to take a crack at it.

The only two services I can think of that really make sense are a way to force clear a block and a night mode toggle.

The propose I could think of for force clearing a block condition is when an automation needs to interact with the sensor, but not override it. I.E.: I want my porch light on dim when the door is open but I want the motion sensor to be able to go to full brightness when motion is detected. The automation for the door sensor could turn the light on low when the door is opened, then clear the block so the entity controller can still trigger if motion is detected. (Recovery of the dim setting would be more steps but I think doable with the current capabilities.)

The night mode toggle would just be a way to move into/out of night mode with automations instead of a fixed/semi-fixed time table, this would probable require a schema change to signify that the start and stop times are not required, probably via an advanced: true setting for night mode?

example service calls.

```yaml
service: entity_controller.clear_block
data:
  entity_id: entity_controller.porch
```
if the controller is not blocked then this should do nothing.

```yaml
service: entity_controller.set_night_mode
data:
  entity_id: entity_controller.porch
  night_mode: on
```
if the entity controller doesn't support night mode or is already in the wanted state, this should do nothing.

I have not looked at the code much yet so I could be making bad assumptions about how these could be implemented, but I wanted to get some thoughts on this.

**Comment by xannor** [comment]
I have built out some code that "works for me" in a branch forked from your master. 
https://github.com/danobot/entity-controller/compare/master...xannor:services

I dont know if you want me to go ahead and submit a pull or if there are any requirements or additional information you want first. I dont know I built it against master instead of dev so it may need to be rebased first, but I built against master since I used my live environment to test it.
